### PR TITLE
Add profile option in suggested commands - Issue #28

### PR DESCRIPTION
**Pull request by maeldebon**
Add the --profile option in the commands suggested by the printers
See #28

### CODE REVIEW

File: substra/cli/interface.py [hunk 1]
```diff
@@ -565,6 +565,7 @@ def get(ctx, asset_name, asset_key, expand, output_format, config, profile, user
     # method must exist in sdk
     method = getattr(client, f'get_{asset_name.lower()}')
     res = method(asset_key)
+    res['profile'] = profile
```
Comment (samlesu) [inline substra/cli/interface.py]: I would recommend to not add the profile to the results returned by the SDK / server. 

It is not part of the data we want to display to the user, it is more something related to how this should be displayed.

This could be added as an option the print command?

Comment (maeldebon) [reply substra/cli/interface.py]: I changed that in the last commit, that's better indeed! 🙂

File: substra/cli/printers.py [hunk 2]
```diff
@@ -167,30 +167,35 @@ def _get_list_fields(self):
     def _get_single_fields(self):
         return (self.key_field, ) + self.single_fields
 
-    def print_download_message(self, item):
+    def print_with_profile(self, profile):
```
Comment (jmorel) [inline substra/cli/printers.py]: In this module, all methods starting with `print_` actually print something. You could rename it to `get_profile_arg` to make its purpose crystal clear.

File: substra/cli/printers.py [hunk 3]
```diff
@@ -167,30 +167,35 @@ def _get_list_fields(self):
...
+        return ''
+
+    def print_download_message(self, item, profile=None):
         if self.download_message:
             key_value = self.key_field.get_value(item)
-            print()
-            print(self.download_message)
-            print(f'\tsubstra download {self.asset_name} {key_value}')
+            profile_arg = self.print_with_profile(profile)
+            print('\n' + self.download_message)
```
Comment (jmorel) [inline substra/cli/printers.py]: Why the change from `print()` to `print("\n")`? It's almost the same, but now we have sometimes `\n` and sometimes `print()` in the same file.

File: substra/cli/printers.py [hunk 4]
```diff
@@ -167,30 +167,35 @@ def _get_list_fields(self):
...
 
-    def print_description_message(self, item):
+    def print_description_message(self, item, profile=None):
         if self.has_description:
             key_value = self.key_field.get_value(item)
-            print()
-            print(f'Display this {self.asset_name}\'s description:')
-            print(f'\tsubstra describe {self.asset_name} {key_value}')
+            profile_arg = self.print_with_profile(profile)
+            print('\n' + f'Display this {self.asset_name}\'s description:')
```
Comment (jmorel) [inline substra/cli/printers.py]: No need to concatenate strings here, you can write `f'\nDisplay etc.'`.

File: substra/cli/printers.py [hunk 5]
```diff
@@ -237,15 +242,20 @@ class ObjectivePrinter(AssetPrinter):
     )
     download_message = 'Download this objective\'s metric:'
 
-    def print_leaderboard_message(self, item):
+    def print_with_profile(self, profile):
+        if profile and profile != 'default':
+            return f'--profile {profile}'
+        return ''
```
Comment (jmorel) [inline substra/cli/printers.py]: Since the `AlgoPrinter` class inherits from `AssetPrinter` it automatically gets the `print_with_profile` method, no need to override it.

File: substra/cli/printers.py (hunk 3 above)
Comment (jmorel) [reply substra/cli/printers.py]: Another solution would be to include the `\n` at the beginning of the download_message string.

File: substra/cli/printers.py (hunk 2 above)
Comment (maeldebon) [reply substra/cli/printers.py]: That's true, I'll change that! 🙂

File: substra/cli/printers.py (hunk 3 above)
Comment (maeldebon) [reply substra/cli/printers.py]: I think that this is cleaner this way, instead of having an empty print. Adding the `\n` at the beginning of the download_message string means that you have to do that for all of these messages. So for me this is easier (and maybe cleaner) to just put only one `\n` when the message is displayed 😃

File: substra/cli/printers.py (hunk 4 above)
Comment (maeldebon) [reply substra/cli/printers.py]: Yes, I'll change that

File: substra/cli/printers.py (hunk 5 above)
Comment (maeldebon) [reply substra/cli/printers.py]: That's true, I remove it! 🙂

File: substra/cli/printers.py (hunk 3 above)
Comment (jmorel) [reply substra/cli/printers.py]: Sure why not. All my arguments were going to be about taste and not exactly productive. Let's go with it!
